### PR TITLE
Feat custom infinite scroll content

### PR DESCRIPTION
**Pull request by xiaohuoni**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: src/components/infinite-scroll/index.tsx [hunk 1]
```diff
@@ -60,21 +74,24 @@ const InfiniteScroll = withDefaultProps({
...
     >
-      {props.hasMore ? (
-        <>
-          <span>加载中</span>
-          <Loading size='small' />
-        </>
-      ) : (
-        <span>没有更多了</span>
-      )}
+      {hasChilds &&
```
Comment (awmleer) [inline src/components/infinite-scroll/index.tsx:86]: 为啥要加 `hasChilds` 的判断？首屏加载的时候保留加载中的指示器应该是很正常的

File: src/components/infinite-scroll/demos/content.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    <>
+      <List>
+        {data.map((item, index) => (
+          <List.Item key={index}>{item}</List.Item>
+        ))}
+      </List>
+      <InfiniteScroll loadMore={loadMore} hasMore={hasMore}>
+        <InfiniteScrollContent />
+      </InfiniteScroll>
```
Comment (awmleer) [inline src/components/infinite-scroll/demos/content.tsx:66]: 直接原封不动地 children 是不是就可以了？`hasMore` 这个状态是外部可以拿到的，`InfiniteScroll` 自动地注入感觉反而有些隐晦了……

```tsx
<InfiniteScroll loadMore={loadMore} hasMore={hasMore}>
  {hasMore ? (
    <>
      <Loading size='small' />
      <span>Loading...</span>
    </>
  ) : (
    <span>--- 我是有底线的 ---</span>
  )}
</InfiniteScroll>
```

Comment (xiaohuoni) [reply src/components/infinite-scroll/demos/content.tsx:66]: 也可以，我改下

File: src/components/infinite-scroll/index.tsx (hunk 1 above)
Comment (xiaohuoni) [reply src/components/infinite-scroll/index.tsx:86]: 嗯，有自定义的时候，首屏加载也要用自定义的，所以这里加了判断

**Review by awmleer — APPROVED** [review]

**Comment by xiaohuoni** [comment]
> 搞了搞了…… [1b8ffdf](https://github.com/ant-design/ant-design-mobile/commit/1b8ffdfc16c88e690e18f6e9b4d315bce75b50e2)

行尾分号，能不能加上，（难受）

**Comment by awmleer** [comment]
> > 搞了搞了…… [1b8ffdf](https://github.com/ant-design/ant-design-mobile/commit/1b8ffdfc16c88e690e18f6e9b4d315bce75b50e2)
> 
> 行尾分号，能不能加上，（难受）

不能……

**Comment by xiaohuoni** [comment]
> > > 搞了搞了…… [1b8ffdf](https://github.com/ant-design/ant-design-mobile/commit/1b8ffdfc16c88e690e18f6e9b4d315bce75b50e2)
> > 
> > 
> > 行尾分号，能不能加上，（难受）
> 
> 不能……

我要去打小报告，你们一个部门用两套规范（哼）

**Pull request merged**